sdk/keyvault/azure-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateClientBuilder.java

```diff
@@ -0,0 +1,259 @@
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.implementation.annotation.ServiceClientBuilder;
+import com.azure.core.implementation.http.policy.spi.HttpPolicyProviders;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.configuration.Configuration;
+import com.azure.core.util.configuration.ConfigurationManager;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of the {@link CertificateAsyncClient certificate async client} and {@link CertificateClient certificate sync client},
+ * by calling {@link CertificateClientBuilder#buildAsyncClient() buildAsyncClient} and {@link CertificateClientBuilder#buildClient() buildClient} respectively
+ * It constructs an instance of the desired client.
+ *
+ * <p> The minimal configuration options required by {@link CertificateClientBuilder} to build {@link CertificateAsyncClient}
+ * are {@link String endpoint} and {@link TokenCredential credential}. </p>
+ *
+ * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.instantiation}
+ *
+ * <p>The {@link HttpLogDetailLevel log detail level}, multiple custom {@link HttpLoggingPolicy policies} and custom
+ * {@link HttpClient http client} can be optionally configured in the {@link CertificateClientBuilder}.</p>
+
+ * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.withhttpclient.instantiation}
+ *
+ * <p>Alternatively, custom {@link HttpPipeline http pipeline} with custom {@link HttpPipelinePolicy} policies and {@link String endpoint}
+ * can be specified. It provides finer control over the construction of {@link CertificateAsyncClient} and {@link CertificateClient}</p>
+ *
+ * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.pipeline.instantiation}
+ *
+ * <p> The minimal configuration options required by {@link CertificateClientBuilder certificateClientBuilder} to build {@link CertificateClient}
+ * are {@link String endpoint} and {@link TokenCredential credential}. </p>
+ *
+ * {@codesnippet com.azure.security.keyvault.certificates.certificateclient.instantiation}
+ *
+ * @see CertificateAsyncClient
+ * @see CertificateClient
+ */
+@ServiceClientBuilder(serviceClients = {CertificateClient.class, CertificateAsyncClient.class})
+    public final class CertificateClientBuilder {
+    private final ClientLogger logger = new ClientLogger(CertificateClientBuilder.class);
+
+    private final List<HttpPipelinePolicy> policies;
+    private TokenCredential credential;
+    private HttpPipeline pipeline;
+    private URL endpoint;
+    private HttpClient httpClient;
+    private HttpLogDetailLevel httpLogDetailLevel;
+    private final RetryPolicy retryPolicy;
```

Comment:
Is the `retryPolicy` not configurable?

Reply:
for that user will need to pass in their custom pipeline with their policies added, I believe.
This policy variable doesn't have a setter.